pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -2,12 +2,15 @@
 import hashlib
 import logging
 import os
+import requests
```

Comment:
The issue with using requests is that it doesn't work w/ the various configuration settings on the Remote, e.g. proxy configurations, ssl, etc. We need to perform all downloading with the pulpcore provided downloaders generated from the remote. I can show an example of this, just lmk.

Reply:
it isn't for download, it is just for checking if the valid URL ends with `.treeinfo` or `treeinfo`

Reply:
I see. My concern is picking up requests as a dependency when we don't really need it. aiohttp provides a similar functionality from the yarl library and we already have that as a dependency. To avoid picking up the dependency could something from yarl be used? https://yarl.readthedocs.io/en/latest/